pkg/infra/usagestats/usage_stats.go

```diff
@@ -186,6 +188,21 @@ func (uss *UsageStatsService) GetUsageReport() (UsageReport, error) {
 	return report, nil
 }
 
+func (uss *UsageStatsService) registerExternalMetrics(metrics map[string]interface{}) {
+	for name, fn := range uss.externalMetrics {
+		result, err := fn()
+		if err != nil {
+			metricsLogger.Error("Failed to fetch external metric", "name", name, "error", err)
+			continue
+		}
+		metrics[name] = result
+	}
+}
+
+func (uss *UsageStatsService) AddMetric(name string, fn MetricFunc) {
```

Comment:
Maybe worth naming it `AddExternalMetric`? 🤔

Reply:
My thinking was that the fact that those metrics are "external" is an implementation detail, which is relevant only to the usage_stats service/package. From a client perspective, you just add a "metric". This also assumes future compatibility, when we decide to not have hardcoded metrics here anymore, thus the "external" concept goes away. wdyt ?

Reply:
Makes sense. Thanks for explaining.